neqo-client/src/main.rs

```diff
@@ -123,135 +121,113 @@ impl KeyUpdateState {
+    #[arg(name = "decoder-table-size", long, default_value = "16384")]
     max_table_size_decoder: u64,
 
-    #[structopt(name = "max-blocked-streams", short = "b", long, default_value = "10")]
+    #[arg(name = "max-blocked-streams", short = 'b', long, default_value = "10")]
     max_blocked_streams: u16,
 
-    #[structopt(name = "max-push", short = "p", long, default_value = "10")]
+    #[arg(name = "max-push", short = 'p', long, default_value = "10")]
     max_concurrent_push_streams: u64,
 
-    #[structopt(name = "use-old-http", short = "o", long)]
+    #[arg(name = "use-old-http", short = 'o', long)]
     /// Use http 0.9 instead of HTTP/3
     use_old_http: bool,
 
-    #[structopt(name = "download-in-series", long)]
+    #[arg(name = "download-in-series", long)]
     /// Download resources in series using separate connections.
     download_in_series: bool,
 
-    #[structopt(name = "concurrency", long, default_value = "100")]
+    #[arg(name = "concurrency", long, default_value = "100")]
     /// The maximum number of requests to have outstanding at one time.
     concurrency: usize,
 
-    #[structopt(name = "output-read-data", long)]
+    #[arg(name = "output-read-data", long)]
     /// Output received data to stdout
     output_read_data: bool,
 
-    #[structopt(name = "qlog-dir", long)]
+    #[arg(name = "qlog-dir", long, value_parser=clap::value_parser!(PathBuf))]
     /// Enable QLOG logging and QLOG traces to this directory
     qlog_dir: Option<PathBuf>,
 
-    #[structopt(name = "output-dir", long)]
+    #[arg(name = "output-dir", long, value_parser=clap::value_parser!(PathBuf))]
     /// Save contents of fetched URLs to a directory
     output_dir: Option<PathBuf>,
 
-    #[structopt(name = "qns-test", long)]
+    #[arg(name = "qns-test", long)]
     /// Enable special behavior for use with QUIC Network Simulator
     qns_test: Option<String>,
 
-    #[structopt(short = "r", long)]
+    #[arg(short = 'r', long)]
     /// Client attempts to resume by making multiple connections to servers.
     /// Requires that 2 or more URLs are listed for each server.
     /// Use this for 0-RTT: the stack always attempts 0-RTT on resumption.
     resume: bool,
 
-    #[structopt(name = "key-update", long)]
+    #[arg(name = "key-update", long)]
     /// Attempt to initiate a key update immediately after confirming the connection.
     key_update: bool,
 
-    #[structopt(short = "c", long, number_of_values = 1)]
+    #[arg(short = 'c', long, number_of_values = 1)]
     /// The set of TLS cipher suites to enable.
     /// From: TLS_AES_128_GCM_SHA256, TLS_AES_256_GCM_SHA384, TLS_CHACHA20_POLY1305_SHA256.
     ciphers: Vec<String>,
 
-    #[structopt(name = "ech", long)]
+    #[arg(name = "ech", long, value_parser=decode_hex)]
     /// Enable encrypted client hello (ECH).
     /// This takes an encoded ECH configuration in hexadecimal format.
     ech: Option<HexArg>,
```

Comment:
```suggestion
    ech: Option<Vec<u8>>,
```

This and delete the unused HexArg.